out_opentelemetry: Add docs for http2 and grpc options (#1521)
fluent/fluent-bit-docs: pipeline/outputs/opentelemetry.md

| http_passwd          | Basic Auth Password. Requires HTTP_user to be set            |           |
| port                 | TCP port of the target HTTP Server                           | 80        |
| proxy                | Specify an HTTP Proxy. The expected format of this value is `http://HOST:PORT`. Note that HTTPS is **not** currently supported. It is recommended not to set this and to configure the [HTTP proxy environment variables](https://docs.fluentbit.io/manual/administration/http-proxy) instead as they support both HTTP and HTTPS. |           |

> | http2 | It defines whether HTTP/2 protocol is enabled. It also supports `force` which will force http2 over a plaintext connection. | On |

Reviewer: ```suggestion
| http2 | Defines whether HTTP/2 protocol is enabled. This setting also supports the `force` option, which forces HTTP/2 over a plaintext connection. | On |
```

> | grpc | Enable gRPC over HTTP/2. It doesn't work for other HTTP versions. | off |

Reviewer: ```suggestion
| grpc | Enables gRPC over an HTTP/2 connection. This setting applies to HTTP/2 only. | off |
```
